### PR TITLE
AVRO-3212: fix specification to allow doc in fixed types

### PR DESCRIPTION
**Pull request by radai-rosenblatt**
Make sure you have checked _all_ steps below.

### Jira

https://issues.apache.org/jira/browse/AVRO-3212

- [x] My PR addresses the following [Avro Jira](https://issues.apache.org/jira/browse/AVRO/) issues and references them in the PR title. For example, "AVRO-1234: My Avro PR"
  - https://issues.apache.org/jira/browse/AVRO-XXX
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

this is just a text change in the docs

### Commits

- [x] My commits all reference Jira issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](https://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain Javadoc that explain what it does

### CODE REVIEW

**Review by martin-g — APPROVED** [review]

File: doc/src/content/xdocs/spec.xml
Comment (martin-g) [inline doc/src/content/xdocs/spec.xml:260]: A few questions not directly related to this PR: 
1) Should we add `(optional)` for namespace too ?
2) Why it uses `<em>` instead of `<code>` like all other attributes ?

Comment (radai-rosenblatt) [reply doc/src/content/xdocs/spec.xml]: change <em> to <code> for namespace - at least I cant think of a reason why it's different?
i agree namespace should be marked optional (it is. it is also ignored is name has dots in it), but will leave that to a committer to decide

Comment (martin-g) [reply doc/src/content/xdocs/spec.xml]: You have changed two occurrences of `<em>` but there are two more. IMO, if we do this change then we should do it for all of the occurrences.

**Review by RyanSkraba — APPROVED** [review]
Thanks so  much for the cleanup!  I agree that the formatting of the Avro spec leaves a lot to be desired. :D  There's AVRO-2175 for a general "refresh", and I'll add a comment there about formatting the Avro specification.

I'm not too worried about the weird `<em>` tags outside of this change until then, let's get this merged!

**Pull request merged**
